src/OSmOSE/cluster/audio_reshaper.py

```diff
@@ -1,630 +1,355 @@
+            + np.arange(0, segment_size * segment_sample_rate) / segment_sample_rate
+        )
 
-                # t += 1
-                audio_data = previous_audio_data
-
-                if not merge_files and len(audio_data) < chunk_size * sample_rate:
-                    previous_audio_data = np.empty(0)
-                    match last_file_behavior:
-                        case "truncate":
-                            output = audio_data
-                            audio_data = []
-                        case "pad":
-                            fill = np.zeros(
-                                (chunk_size * sample_rate) - len(audio_data)
-                            )
-                            output = np.concatenate((audio_data, fill))
-                            audio_data = []
-                        case "discard":
-                            audio_data = []
-                            break
-
-                    outfilename = output_dir_path.joinpath(
-                        f"{datetime.strftime(timestamp, '%Y-%m-%dT%H:%M:%S').replace('-','_').replace(':','_').replace('.','_').replace('+','_')}.{fmt}"
-                    )
-                    result.append(outfilename.name)
+        len_sig = 0
+        audio_data_sum = audio_data.sum()
+        for index, row in file_metadata.iterrows():
 
-                    list_seg_name.append(outfilename.name)
-                    list_seg_timestamp.append(
-                        datetime.strftime(timestamp, "%Y-%m-%dT%H:%M:%S.%f%z")
-                    )
+            file_datetime_begin = row["timestamp"]
+            file_datetime_end = row["timestamp"] + pd.Timedelta(seconds=row["duration"])
 
-                    timestamp_list.append(
-                        datetime.strftime(timestamp, "%Y-%m-%dT%H:%M:%S.%f%z")
-                    )
-                    timestamp += timedelta(seconds=len(output))
-
-                    sf.write(
-                        outfilename,
-                        output,
-                        sample_rate,
-                        format=fmt,
-                        subtype=file_type,
-                    )
-                    os.chmod(outfilename, mode=FPDEFAULT)
+            # check if original audio file overlaps with the current segment
+            if (
+                file_datetime_end > segment_datetime_begin
+                and file_datetime_begin < segment_datetime_end
+            ):
 
-                    pad_text = (
-                        f"Padded with {fill.size // sample_rate} seconds."
-                        if last_file_behavior == "pad" and fill.size > 0
-                        else ""
+                start_offset = (
+                    0
+                    if file_datetime_begin >= segment_datetime_begin
+                    else int(
+                        (segment_datetime_begin - file_datetime_begin).total_seconds()
+                        * orig_sr
                     )
```

Comment:
I think this can be simplified:
```suggestion
                start_offset = max(0, (segment_datetime_begin - file_datetime_begin).total_seconds()))  * orig_sr
```

Reply:
`start_offset = int(max(0, (segment_datetime_begin - file_datetime_begin).total_seconds()))  * orig_sr`

Reply:
Maybe even
```py
start_offset = int(max(0., (segment_datetime_begin - file_datetime_begin).total_seconds()))  * orig_sr
```
Otherwise you could have a warning due to `max()` having both an int and a float as parameters.